From 7f4238e53177325e416ae0dfa4d5a42db1f53be3 Mon Sep 17 00:00:00 2001
From: Francisco Boni Neto <boboniboni@gmail.com>
Date: Sun, 10 Jul 2022 09:42:36 -0300
Subject: [PATCH 34/36] novector

---
 arch/x86/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/Makefile b/arch/x86/Makefile
index 9e6094b9e..a0328a9e1 100644
--- a/arch/x86/Makefile
+++ b/arch/x86/Makefile
@@ -66,7 +66,7 @@ export BITS
 #
 #    https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53383
 #
-KBUILD_CFLAGS += -mno-sse -mno-mmx -mno-sse2 -mno-3dnow -mno-avx
+KBUILD_CFLAGS += -mno-sse -mno-mmx -mno-sse2 -mno-3dnow -mno-avx -mno-avx2 -O3 -fno-tree-vectorize -fno-tree-loop-vectorize -fno-tree-slp-vectorize -mpopcnt
 
 ifeq ($(CONFIG_X86_KERNEL_IBT),y)
 #
-- 
2.36.0

